From: Bhavna Sarathy <bnagendr@redhat.com>
Date: Thu, 5 Aug 2010 18:03:57 -0400
Subject: [edac] amd64_edac: fix forcing module load/unload
Message-id: <20100805180939.6566.74778.sendpatchset@localhost.localdomain>
Patchwork-id: 27418
O-Subject: [RHEL5.6 PATCH 4/10] amd64_edac - Fix forcing module load/unload
Bugzilla: 568576
RH-Acked-by: Mauro Carvalho Chehab <mchehab@redhat.com>

>From 99f68258a800e0bf5f82d7582478cfd89a8c149c Mon Sep 17 00:00:00 2001
From: Borislav Petkov <borislav.petkov@amd.com>
Date: Mon, 2 Aug 2010 23:06:22 +0200
Subject: [PATCH] amd64_edac: fix forcing module load/unload

(Upstream commit 43f5e68733cfe8bed3c30b5c14c4993dffb29766)

Clear the override flag after force-loading the module.

Signed-off-by: Borislav Petkov <borislav.petkov@amd.com>

diff --git a/drivers/edac/amd64_edac.c b/drivers/edac/amd64_edac.c
index c8642b2..49b479d 100644
--- a/drivers/edac/amd64_edac.c
+++ b/drivers/edac/amd64_edac.c
@@ -2532,11 +2532,9 @@ static int amd64_check_ecc_enabled(struct amd64_pvt *pvt)
                if (!ecc_enable_override) {
                        amd64_printk(KERN_WARNING, "%s", ecc_warning);
                        return -ENODEV;
-               }
-       } else
-               /* CLEAR the override, since BIOS controlled it */
-               ecc_enable_override = 0;
-
+	       }
+	       ecc_enable_override = 0;
+       }
        return 0;
 }
 
